src/zarr/indexing.py

```diff
@@ -1206,8 +1196,8 @@ def pop_fields(selection: SelectionWithFields) -> tuple[Fields | None, Selection
         return fields, selection
 
 
-def make_slice_selection(selection: Any) -> list[int | slice]:
-    ls: list[int | slice] = []
+def make_slice_selection(selection: Any) -> list[IntOrSlice]:
```

Comment:
question for people who know array indexing better than me: is there any reason for this function to return `x: int`, when we can express the same thing with `slice(x, x + 1, 1)`?

Reply:
I think you are right, it should only return `list[slice]`. Is that what you mean?

Reply:
No, I think the annotation is correct for this function, but I am curious more broadly about _why_ this function returns ints, as opposed to turning ints into slices